spec/package-manager-spec.coffee

```diff
@@ -286,6 +290,34 @@ describe "PackageManager", ->
             expect(addErrorHandler.callCount).toBe 1
             expect(addErrorHandler.argsForCall[0][0].message).toContain("Failed to load the package-with-invalid-settings package settings")
 
+    describe "when the package metadata includes `activationHooks`", ->
+      [mainModule, promise] = []
+
+      beforeEach ->
+        jasmine.attachToDOM(workspaceElement)
+        mainModule = require './fixtures/packages/package-with-activation-hooks/index'
+        spyOn(mainModule, 'activate').andCallThrough()
+        spyOn(Package.prototype, 'requireMainModule').andCallThrough()
+
+        promise = atom.packages.activatePackage('package-with-activation-hooks')
+
+      it "defers requiring/activating the main module until an triggering of an activation hook occurs", ->
```

Comment:
What do you think about making some mock assertions here? Did you want to use the `requireMainModule` mock?